inventory/base_inventory.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python
+# Copyright 2018, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import copy
+import json
+import os
+
+
+class MaasInventory(object):
```

Comment:
@npawelek @cloudnull I am unsure if there are immediate plans to make more classes (different inventory types) here? I tend to agree with Brian if this will be all we are going to need, but if there are plans to add more after this, I can see why the base class would be necessary.

Reply:
@briancurtin Tony slacked me this: 
>We used subclass because we might want to use the same dynamic inventory mechanism to other platform like ceph, or maybe SUSE based rpc clouds
I will make it an `ABC`